docs/book/content/types/unions.md

```diff
@@ -1,75 +1,197 @@
-# Unions
+Unions
+======
 
-From a server's point of view, GraphQL unions are similar to interfaces: the
-only exception is that they don't contain fields on their own.
+From a server's point of view, [GraphQL unions][1] are similar to interfaces: the only exception is that they don't contain fields on their own.
+
+For implementing [GraphQL union][1] Juniper provides:
+- `#[derive(GraphQLUnion)]` macro for enums and structs;
+- `#[graphql_union]` for traits.
 
-In Juniper, the `graphql_union!` has identical syntax to the
-[interface macro](interfaces.md), but does not support defining
-fields. Therefore, the same considerations about using traits,
-placeholder types, or enums still apply to unions. For simple
-situations, Juniper provides `#[derive(GraphQLUnion)]` for enums.
 
-If we look at the same examples as in the interfaces chapter, we see the
-similarities and the tradeoffs:
 
-## Traits
 
-### Downcasting via accessor methods
+## Enums
+
+Most of the time, we need just a trivial and straightforward Rust enum to represent a [GraphQL union][1].
 
 ```rust
-#[derive(juniper::GraphQLObject)]
+use derive_more::From;
+use juniper::{GraphQLObject, GraphQLUnion};
+
+#[derive(GraphQLObject)]
 struct Human {
     id: String,
     home_planet: String,
 }
 
-#[derive(juniper::GraphQLObject)]
+#[derive(GraphQLObject)]
 struct Droid {
     id: String,
     primary_function: String,
 }
 
-trait Character {
-    // Downcast methods, each concrete class will need to implement one of these
-    fn as_human(&self) -> Option<&Human> { None }
-    fn as_droid(&self) -> Option<&Droid> { None }
+#[derive(From, GraphQLUnion)]
+enum Character {
+    Human(Human),
+    Droid(Droid),
 }
+#
+# fn main() {}
+```
 
-impl Character for Human {
-    fn as_human(&self) -> Option<&Human> { Some(&self) }
+
+### Ignoring enum variants
+
+In some rare situations we may want to omit exposing enum variant in GraphQL schema.
```

Comment:
@LegNeato hmm... I tried to use `we` instead `you` everywhere to be "on the reader's side" during this "journey", rather than "command him" 🙃

If it's not OK, I'll change the style to use `you` everywhere.

Reply:
Ok, that works for me.